controllers/serviceclaim_controller.go

```diff
@@ -488,7 +489,41 @@ func (r *ServiceClaimReconciler) DeleteServiceBindingsAndSecret(
 
 // SetupWithManager sets up the controller with the Manager.
 func (r *ServiceClaimReconciler) SetupWithManager(mgr ctrl.Manager) error {
+
+	genPred := predicate.GenerationChangedPredicate{}
+	reconcileOnRegisteredServiceUpdate := func(ctx context.Context, a client.Object) []reconcile.Request {
+		l := log.FromContext(ctx)
+		rs, ok := a.(*v1alpha1.RegisteredService)
+		if !ok {
+			l.Info("error parsing object to RegisteredService when mapping to ServiceClaim reconciliation trigger", "object", a)
+			return []reconcile.Request{}
+		}
+		if rs.Status.State != `Claimed` {
+			l.Info(fmt.Sprintf("Registered service %v is unclaimed, no need to reconcile service claim", rs.Name))
+			return []reconcile.Request{}
+		}
+		serviceclaims := &v1alpha1.ServiceClaimList{}
+		opts := &client.ListOptions{}
+		if err := r.List(ctx, serviceclaims, opts); err != nil {
+			l.Error(err,
+				"unable to list the ServiceClaims and reconcile for Registered Service Updates",
+				"RegisteredService", rs.Name)
+			return []reconcile.Request{}
+		}
+		reply := make([]reconcile.Request, 0, len(serviceclaims.Items))
+		for _, sc := range serviceclaims.Items {
+			if !reflect.DeepEqual(sc.Status, v1alpha1.ServiceClaimStatus{}) && sc.Status.State == "Resolved" && sc.Status.RegisteredService == rs.Name {
```

Comment:
I think we can simplify this check. WDYT?

```suggestion
			if sc.Status.State == "Resolved" && sc.Status.RegisteredService == rs.Name {
```

Reply:
Can the sc.Status be empty?.. It would result in panic in that case.

Reply:
we have panic only if sc.Status is nil, but it is not possible as it is not a pointer

```go
type ServiceClaim struct {
	metav1.TypeMeta   `json:",inline"`
	metav1.ObjectMeta `json:"metadata,omitempty"`

	Spec   ServiceClaimSpec   `json:"spec,omitempty"`
	Status ServiceClaimStatus `json:"status,omitempty"`
}
```